// python3 -m grpc_tools.protoc -I protos --python_out=. --grpc_python_out=. protos/chat.proto

syntax = "proto3";

enum ServerOperation {
    SUCCESS = 0;
    FAILURE = 1;
    ACCOUNT_ALREADY_EXISTS = 2;
    ACCOUNT_DOES_NOT_EXIST = 3;
    LIST_OF_ACCOUNTS = 4;
    LIST_OF_MESSAGES = 5;
    NO_MESSAGES = 6;
    MESSAGES_EXIST = 7;
}

message ClientMessage {
    string info = 1;
  }

message ServerMessage {
    ServerOperation operation = 1;
    string info = 2;
}

service ChatService {

    rpc LoginClient (ClientMessage) returns (ServerMessage) {}
    rpc CreateAccountClient (ClientMessage) returns (ServerMessage) {}
    rpc DeleteAccountClient (ClientMessage) returns (ServerMessage) {}
    rpc ListAccountClient (ClientMessage) returns (ServerMessage) {}
    rpc SendMessageClient (ClientMessage) returns (ServerMessage) {}
    rpc ViewMessageClient (ClientMessage) returns (ServerMessage) {}
    rpc LogoutClient (ClientMessage) returns (ServerMessage) {}
    rpc CheckIncomingMessagesClient (ClientMessage) returns (ServerMessage) {}

}